### PR TITLE
rm router logits Improve TTOP 3ms

### PR DESCRIPTION
**Pull request by ttanzhiqiang**
### What this PR does / why we need it?

The previous code is
router_logits, _ = self.gate(hidden_states)
hidden_states = get_dp_group().all_gather(hidden_states, 0)
router_logits = get_dp_group().all_gather(router_logits, 0)
I want to change the two all_gathers to one, reduce one all_gather communication, and make it
hidden_states = get_dp_group().all_gather(hidden_states, 0)
router_logits, _ = self.gate(hidden_states)

### How was this patch tested?

bash examples/run_dp_attention_etp16.sh
bash examples/run_dp_attention_etp16_benmark.sh

gsm8k accuracy verification
截屏2025-06-24 21 53 24

### CODE REVIEW

File: vllm_ascend/ops/fused_moe.py [hunk 1]
```diff
@@ -1137,6 +1137,7 @@ def __init__(
         self.activation = activation
         self.log2phy = None
         self.global_redundant_expert_num = 0
+        self.rm_router_logits = envs_ascend.VLLM_ASCEND_RM_ROUTER_LOGITS
```
Comment (ApsarasX) [inline vllm_ascend/ops/fused_moe.py]: It is recommended to enable this feature only during the decode phase.

Comment (ttanzhiqiang) [reply vllm_ascend/ops/fused_moe.py]: The prefill phase should also be optimized, as long as there are two all_gather communications

File: vllm_ascend/envs.py [hunk 2]
```diff
@@ -121,6 +121,9 @@
     # value to False to disable the optimized model.
     "USE_OPTIMIZED_MODEL":
     lambda: bool(int(os.getenv('USE_OPTIMIZED_MODEL', '1'))),
+    # Remove the two communications of get_dp_group().all_gather and change it to one, and do gate after the communication
+    "VLLM_ASCEND_RM_ROUTER_LOGITS":
+    lambda: int(os.getenv("VLLM_ASCEND_RM_ROUTER_LOGITS", 0)),
```
Comment (wangxiyuan) [inline vllm_ascend/envs.py]: from Q3, we'll be careful to add more configuration. please remove it to enable rm_router_logits by default.

Comment (ttanzhiqiang) [reply vllm_ascend/envs.py]: This is only valid in the FusedMoEState.AllGather solution. If other models use gate externally and rm_router_logits internally, an error will be reported, such as deepseek_dbo/qwen3/qwen2

Comment (Yikun) [reply vllm_ascend/envs.py]: Agree, could we enable this in some cases automatically, because it's difficult to let users know which models should enable this env or not.

Otherwise, LGTM

Comment (ttanzhiqiang) [reply vllm_ascend/envs.py]: In theory, this solution is only applicable to AllGather and AllGatherEP, because in the dp scenario, the previous operation was gate + two communications, and now it is changed to one communication + gate operation, which can save some communication time. In theory, all moe AllGather and AllGatherEP solutions can follow this logic, but now other moe models (qwen3-235b) dp solutions are not adjusted, so use the switch to control it to prevent code errors.

Comment (wangxiyuan) [reply vllm_ascend/envs.py]: If it's not common, I prefer not to merge, we can wait more.

Or, if we can add more logic check instead of env var, i'm  fine as well.

Comment (ttanzhiqiang) [reply vllm_ascend/envs.py]: ok, currently m is enabled by default in the AllGather, AllGatherEP and NaiveMulticast scenarios of the deepseek model. rm_router_logits is not enabled in other scenarios and models. You can add it later if necessary.

**Comment by ttanzhiqiang** [comment]
update @wangxiyuan @Yikun

**Review by wangxiyuan — APPROVED** [review]
